### PR TITLE
[common-4] Replace converter classes of Security package with Mapstruct

### PR DESCRIPTION
**Pull request by vitaliibredun**
Screenshot 2023-08-19 at 16 25 37
1. Task's name: 
[common-4] Replace converter classes of Security package with Mapstruct

 2. Developer's full name: 
Vitalii Bredun

 3. Link to the PR: 
https://github.com/Sunagatov/Online-Store/pull/86

 4. Link to Documentation: 
https://github.com/Sunagatov/Online-Store/wiki/FR-Payment-Integration

 5. Changes: 
- Added the Mapstruct interfaces to the converter package.
- Deleted the custom mapper classes.
- Added tests for the Mapstruct.
                                                                                                                                                                                          
 6. Tests: 
RegistrationDtoConverterTest is checking the mapping.

 7. Test data: 
In the security package of the test package.

 8. Conclusion: 
Project was built with Apache Maven successfully.

### CODE REVIEW

**Review by Sanya124 — APPROVED** [review]

File: src/test/java/com/zufar/onlinestore/RegistrationDtoConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package com.zufar.onlinestore;
```
Comment (nosypro) [inline src/test/java/com/zufar/onlinestore/RegistrationDtoConverterTest.java:1]: Please remove this test class

**Review by nosypro — CHANGES_REQUESTED** [review]

File: src/main/java/com/zufar/onlinestore/security/converter/RegistrationDtoConverter.java [hunk 2]
```diff
@@ -2,20 +2,10 @@
...
-                .firstName(userRegistrationRequest.firstName())
-                .lastName(userRegistrationRequest.lastName())
-                .username(userRegistrationRequest.username())
-                .email(userRegistrationRequest.email())
-                .password(userRegistrationRequest.password())
-                .build();
-    }
-}
+    UserDto toDto(final UserRegistrationRequest userRegistrationRequest);
+}
```
Comment (nosypro) [inline src/main/java/com/zufar/onlinestore/security/converter/RegistrationDtoConverter.java:11]: Add new empty line at the end of this file

**Review by nosypro — CHANGES_REQUESTED** [review]

**Review by AiratGimaev — APPROVED** [review]
Approved

**Review by Sanya124 — APPROVED** [review]

**Review by Sunagatov — APPROVED** [review]

**Review by vpshvd — APPROVED** [review]

File: src/test/java/com/zufar/onlinestore/RegistrationDtoConverterTest.java (hunk 1 above)
Comment (vitaliibredun) [reply src/test/java/com/zufar/onlinestore/RegistrationDtoConverterTest.java]: done

File: src/main/java/com/zufar/onlinestore/security/converter/RegistrationDtoConverter.java (hunk 2 above)
Comment (vitaliibredun) [reply src/main/java/com/zufar/onlinestore/security/converter/RegistrationDtoConverter.java]: done

**Pull request merged**